tms/models/res_partner.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+# pylint: disable=R8180
+
+
+class Driver(models.Model):
+    _inherit = "res.partner"
+
+    # Flags
+
+    is_driver = fields.Boolean(string="Is a driver")
+    is_external = fields.Boolean()
+    is_training = fields.Boolean()
+
+    # Licences and Certifications
+
+    license_number = fields.Char(string="Driver License Number")
+    license_expiry_date = fields.Date()
+    other_certifications = fields.Text()
+
+    # Vehicles
+    vehicles_ids = fields.Many2many("tms.vehicle", string="Assigned Vehicles")
+
+    # Contact Information
+
+    contact_number = fields.Char()
+    emergency_contact_name = fields.Char()
+    emergency_contact_number = fields.Char()
+
+    # Personal Information
+
+    date_of_birth = fields.Date()
+    gender = fields.Selection(
+        [("male", "Male"), ("female", "Female"), ("other", "Other")],
+    )
+    nationality = fields.Char()
+    identity_document = fields.Binary()
+    identity_document_type = fields.Char()
+
+    notes = fields.Text(string="Notes/Comments")
+
+
+class Location(models.Model):
+    _inherit = "res.partner"
```

Comment:
IMHO, this is an implementation error. 

As Location and Driver classes are sharing the same table (res_partner).

IMHO, those fields should go in :

```suggestion
    _name = "tms.location"
    _inherits = "res.partner"
```

Same for Driver.